src/autoschedulers/adams2019/Cache.h

```diff
@@ -13,6 +13,45 @@ namespace Halide {
 namespace Internal {
 namespace Autoscheduler {
 
```

Comment:
This needs a big-picture overview comment as well as the list of changes below. E.g. there seem to be two types of caching: feature caching and block caching, but I'm still confused about what block caching is from reading the text below. Say what kinds of caching exist, what values are cached, what the key is, and why this saves work.

Reply:
Hopefully addressed by dccd642 ? Let me know what you think.